src/main/java/UseCase/RegistrationManager.java

```diff
@@ -0,0 +1,110 @@
+package UseCase;
+
+import Entities.UserAccount;
+import Entities.UserDatabase;
+
+import java.util.ArrayList;
+
+public class RegistrationManager implements RegistrationInputBoundary{
+
+    public RegistrationManager() {}
+
+    public boolean isValidRegistration(UserDatabase database,
+                                       String username,
+                                       String full_name,
+                                       int age,
+                                       String pronouns,
+                                       String country, String province, String city,
+                                       String gender,
+                                       String sexuality,
+                                       String interest,
+                                       String password) {
+        if (isValidUsername(username)) {
+            if (isValidPronouns(pronouns)) {
```

Comment:
This might be smelly code. You might want to find a way to better format this since the indentation can be hard to check for bugs or can be hard to alter.